From 9060cafb608967d06c990303ee30817ee8111d94 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:42:46 -0400
Subject: [infiniband] cxgb4: Always release neigh entry

Message-id: <1410475447-94392-28-git-send-email-svemuri@redhat.com>
Patchwork-id: 93846
O-Subject: [RHEL7.1 PATCH BZ 1124947 027/108] RDMA/cxgb4: Always release neigh entry
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Always release the neigh entry in rx_pkt().

Based on original work by Santosh Rastapur <santosh@chelsio.com>.

Signed-off-by: Steve Wise <swise@opengridcomputing.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit ebf00060c33b9d0946384fa6f440df7ea35a569e)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/cxgb4/cm.c b/drivers/infiniband/hw/cxgb4/cm.c
index 8332994..3629b52 100644
--- a/drivers/infiniband/hw/cxgb4/cm.c
+++ b/drivers/infiniband/hw/cxgb4/cm.c
@@ -3347,13 +3347,13 @@ static int rx_pkt(struct c4iw_dev *dev, struct sk_buff *skb)
   pi = (struct port_info *)netdev_priv(pdev);
   tx_chan = cxgb4_port_chan(pdev);
  }
+ neigh_release(neigh);
  if (!e) {
   pr_err("%s - failed to allocate l2t entry!\n",
          __func__);
   goto free_dst;
  }
 
- neigh_release(neigh);
  step = dev->rdev.lldi.nrxq / dev->rdev.lldi.nchan;
  rss_qid = dev->rdev.lldi.rxq_ids[pi->port_id * step];
  window = (__force u16) htons((__force u16)tcph->window);
-- 
1.7.1